File: Ruby/P4/Irrgarten/fuzzy_player.rb

#encoding:utf-8

=begin

    Autor: Isabel Morro Tabares 

    Funcionalidades de los jugadores Fuzzy

=end

require_relative 'dice'
require_relative 'player'

module Irrgarten
    class FuzzyPlayer < Player

        def initialize (other)
            super(other.number, other.intelligence, other.strength)
            copia(other)
            @name = "Fuzzy " + @name
        end

        def move (direction, valid_moves)
            result = super
            Dice.next_step(result, valid_moves, intelligence)
        end

        def attack
            sum_weapons + Dice.intensity(strength)
        end

        def defensive_energy
            sum_shields + Dice.intensity(intelligence)
        end

    end
end

File: Java/P5/P5/irrgarten/irrgarten/src/irrgarten/FuzzyPlayer.java

package irrgarten;

import java.util.ArrayList;

/**
 * @brief Representa al tipo de jugador FuzzyPlayer
 * @author Isabel  Morro Tabares
 */
public class FuzzyPlayer extends Player {
    
    public FuzzyPlayer (Player other){
        super(other);
        setName("Fuzzy " + other.getName());
    }
    
    @Override
    public Directions move (Directions direction, ArrayList<Directions>  validMoves){
        Directions result = super.move(direction, validMoves);
        return Dice.nextStep(result, validMoves, getIntelligence());
    }
    
    @Override
    public float attack(){
        return sumWeapons() + Dice.intensity(getStrength());
        
    }
    
    @Override
    protected float defensiveEnergy (){
        return sumShields()+ Dice.intensity(getIntelligence());
    }
}

        